Dalamud/Interface/Internal/TextureManager.cs

```diff
@@ -0,0 +1,458 @@
+        path = FormatIconPath(
+            iconId,
+            null,
+            hiRes);
+        if (this.dataManager.FileExists(path))
+            return this.CreateWrap(path, keepAlive);
+        
+        // 4. Regular icon, without language, no hi-res
+        if (hiRes)
+        {
+            path = FormatIconPath(
+                iconId,
+                null,
+                false);
+            if (this.dataManager.FileExists(path))
+                return this.CreateWrap(path, keepAlive);
+        }
+
+        return null;
+    }
+
+    /// <summary>
+    /// Get a texture handle for the texture at the specified path.
+    /// You may only specify paths in the game's VFS.
+    /// </summary>
+    /// <param name="path">The path to the texture in the game's VFS.</param>
+    /// <param name="keepAlive">Prevent Dalamud from automatically unloading this texture to save memory. Usually does not need to be set.</param>
+    /// <returns>Null, if the icon does not exist, or a texture wrap that can be used to render the texture.</returns>
+    public TextureManagerTextureWrap? GetTextureFromGamePath(string path, bool keepAlive)
+    {
+        return !this.dataManager.FileExists(path) ? null : this.CreateWrap(path, keepAlive);
+    }
+    
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        this.framework.Update -= this.FrameworkOnUpdate;
+        
+        Log.Verbose("Disposing {Num} left behind textures.");
+        
+        foreach (var activeTexture in this.activeTextures)
+        {
+            activeTexture.Value.Wrap?.Dispose();
+        }
+        
+        this.activeTextures.Clear();
+    }
+
+    /// <summary>
+    /// Get texture info.
+    /// </summary>
+    /// <param name="path">Path to the texture.</param>
+    /// <param name="refresh">Whether or not the texture should be reloaded if it was unloaded.</param>
+    /// <returns>Info object storing texture metadata.</returns>
+    internal TextureInfo GetInfo(string path, bool refresh = true)
+    {
+        TextureInfo? info;
+        lock (this.activeTextures)
+        {
+            this.activeTextures.TryGetValue(path, out info);
+        }
+
+        if (info == null)
+        {
+            info = new TextureInfo();
+            lock (this.activeTextures)
+            {
+                if (!this.activeTextures.TryAdd(path, info))
+                    Log.Warning("Texture {Path} tracked twice, this might not be an issue", path);
+            }
+        }
+        
+        if (refresh && info.KeepAliveCount == 0)
+            info.LastAccess = DateTime.Now;
+        
+        if (info is { Wrap: not null })
+            return info;
+
+        if (refresh)
+        {
+            byte[]? interceptData = null;
+            this.InterceptTexDataLoad?.Invoke(path, ref interceptData);
+
+            // TODO: Do we also want to support loading from actual fs here? Doesn't seem to be a big deal, collect interest
```

Comment:
We could easily change this API to also support loading images(png) or .tex from the FS. Should that be a goal?

Reply:
Yeah, I think loading in tex from the filesystem would be good if it's simple. I can think of some edge cases where someone has already handled custom resolutions (e.g. loading from a specific Penumbra collection).